helm/vmss-prototype/auto-update.md

```diff
@@ -0,0 +1,243 @@
+# Manual Update
+
+This is a higher level description of the basic functions of the VMSS-Prototype Pattern (Kamino) system.  This goes into deep details as to what you are doing on the machine.
+
+## Example
+
+Below is the canonical way to run vmss-prototype on your cluster using our published Helm Chart to do automatic updates to all VMSS pools in the cluster:
+
+```bash
+$ helm install --repo https://jackfrancis.github.io/kamino/ \
+  update-vmss-model-images \
+  vmss-prototype \
+  --namespace default \
+  --set kamino.scheduleOnControlPlane=true \
+  --set kamino.logLevel=DEBUG \
+  --set kamino.targetVMSS=ALL \
+  --set kamino.auto.lastPatchAnnotation=LatestOSPatch \
+  --set kamino.auto.pendingRebootAnnotation=PendingReboot
+```
+
+
+# Annotated logs from a single VMSS cluster run with automatic mode
+
+This cluster has tools that automatically apply OS patches and do reboots as needed, setting the annotations needed.  There is a PR for the open source [Kured](https://github.com/weaveworks/kured) tool that will have it set these annotations as needed but my test environment did not have that custom build so we used our internal tool.
+
+## First run with effectively the above helm command settings
+
+`vmss-prototype` always logs the command line as it starts for easier debugging.
+```
+2021-01-14T17:08:17.508568931Z CMD: ['/usr/bin/vmss-prototype' '--in-cluster' '--log-level' 'DEBUG' '--log-prefix' 'auto-update' 'auto-update' '--new-updated-nodes' '0' '--grace-period' '5' '--max-history' '3' '--last-patch-annotation' 'LatestOSPatch' '--pending-reboot-annotation' 'PendingReboot' '--minimum-ready-time' '1h' '--minimum-candidates' '1']
+```
+
+Note that at this point, all of the auto-update log lines will have "auto-update" prefixed.  (The timestamps are from kubernetes `kubectl logs --timestamps` and are not ones generated by `vmss-prototype`)
+
+`vmss-prototype` logging into the azure world using the azure.json/node credentials

```

Comment:
To be consistent, let's refer to Azure using a capital "A" everywhere in this do if we're talking about the cloud service. (Obviously for things like `azure.json` we want to use the specific casing for that file reference)

Reply:
Good point.